sccameraview/src/main/java/com/example/sccameraview/BaseCameraView.java

```diff
@@ -53,20 +60,24 @@ public void setCalculatedHeight(int calculatedHeight) {
     }
 
     CamcorderProfile getCamcorderProfile() {
+        if (CamcorderProfile.hasProfile(cameraId, CamcorderProfile.QUALITY_HIGH)) {
+            return CamcorderProfile.get(cameraId, CamcorderProfile.QUALITY_HIGH);
+        }
         if (CamcorderProfile.hasProfile(cameraId, CamcorderProfile.QUALITY_480P)) {
             return CamcorderProfile.get(cameraId, CamcorderProfile.QUALITY_480P);
         }
-
         return CamcorderProfile.get(cameraId, CamcorderProfile.QUALITY_LOW);
     }
 
     abstract void openCamera();
 
+    public abstract void switchCamera();
```

Comment:
It is okay, just add a setter to it, so if the client wants back-facing, currently he must open the normal camera and then switch to back. He should be able to just open the back camera in the first place, by setting that.